### ISSUE TITLE
first viewer of stream never gets latest stream data

### ISSUE THREAD

**Issue by KenanY** [ILiedAboutCake/Rustla2#9]
Steps to reproduce:

  1. Be the first user to begin watching a live Twitch stream
  2. External viewer count is not shown because it has not been fetched server-side yet (this is expected)
  3. Server fetches external viewer count and updates database entry for that stream
  4. Refresh page. External viewer count is still not shown (stream gets removed from db since this was the only user watching?)
  5. Open new tab with the same stream. That second tab will show the external viewer count properly.
  6. First tab can now refresh and get the viewer count.

So, basically, the first user watching a stream will never be able to get an external viewer count, unless they refresh _while another user is watching as well_.

**Comment by blushies** [comment]
I'm not getting this. The stream gets deleted from the DB once all rustlers are gone so when the second tab opens, it doesn't show `viewers`, and when I refresh the first tab doesn't have the info either, as expected

**Comment by KenanY** [comment]
Yup, appears fixed now.
